asapdiscovery/data/utils.py

```diff
@@ -230,36 +238,44 @@ def cdd_to_schema(cdd_csv, out_json=None, out_csv=None, achiral=False):
     df["pIC50_range"] = pic50_range
     semiquant = df["pIC50_range"].astype(bool)
 
-    ### TODO: handle multiple measurements for the same compound
-    ###  (average of stderr for each compound)
-    ci_lower_key = (
-        "ProteaseAssay_Fluorescence_Dose-Response_Weizmann: IC50 "
-        "CI (Lower) (µM)"
-    )
-    ci_upper_key = (
-        "ProteaseAssay_Fluorescence_Dose-Response_Weizmann: IC50 "
-        "CI (Upper) (µM)"
-    )
+    ## Don't need to recompute stderr if it's already there
+    if "pIC50_stderr" not in df.columns:
+        ### TODO: handle multiple measurements for the same compound
+        ###  (average of stderr for each compound)
+        ci_lower_key = (
+            "ProteaseAssay_Fluorescence_Dose-Response_Weizmann: IC50 "
+            "CI (Lower) (µM)"
+        )
+        ci_upper_key = (
+            "ProteaseAssay_Fluorescence_Dose-Response_Weizmann: IC50 "
+            "CI (Upper) (µM)"
+        )
+
+        ## Convert IC50 vals to pIC50 and calculate standard error for  95% CI
+        pic50_stderr = []
+        for _, (ci_lower, ci_upper) in df[
+            [ci_lower_key, ci_upper_key]
+        ].iterrows():
+            ## Cast to float
+            try:
+                ci_lower = float(ci_lower)
+                ci_upper = float(ci_upper)
+            except ValueError:
+                pic50_stderr.append(np.nan)
+                continue
+            if pandas.isna(ci_lower) or pandas.isna(ci_upper):
+                pic50_stderr.append(np.nan)
+            else:
+                ## First convert bounds from IC50 (uM) to pIC50
+                pic50_ci_upper = -np.log10(ci_upper * 10e-6)
```

Comment:
Shouldn't these be reversed, since the smallest IC50 will be the biggest pIC50?
Also, there's a conversion error here: You're using `10e-6`, which is 10 µM. You want `1e-6`.
Perhaps we can use `pint` or some other units library to handle units for us to avoid conversion errors like this?

Reply:
If we want to use some units library, may I suggest using [`openff-units`](https://github.com/openforcefield/openff-units)? That way we can convert to/from openmm units directly, which should be convenient for our purposes.